Source/carthage/Extensions.swift

```diff
@@ -47,6 +47,28 @@ internal func print<T>(object: T) {
 	}
 }
 
+internal class FileHandle {
+	let fileDescriptor: Int32
+	let pointer: UnsafeMutablePointer<FILE>
+
+	init(fileDescriptor fd: Int32) {
+		fileDescriptor = fd
+		pointer = fdopen(fd, "r+")
```

Comment:
I think this should be a class function that returns a `Result`. The error can be from `NSPOSIXErrorDomain` if something goes wrong here.